db/migrate/20231018192110_add_index_to_webauthn_credentials_user_id_nickname.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class AddIndexToWebauthnCredentialsUserIdNickname < ActiveRecord::Migration[7.0]
+  disable_ddl_transaction!
+
+  def up
+    add_index_to_table
+  rescue ActiveRecord::RecordNotUnique
+    remove_duplicates_and_reindex
+  end
+
+  def down
+    remove_index_from_table
+  end
+
+  private
+
+  def remove_duplicates_and_reindex
+    deduplicate_records
+    reindex_records
+  rescue ActiveRecord::RecordNotUnique
+    retry
+  end
+
+  def reindex_records
+    remove_index_from_table
+    add_index_to_table
+  end
```

Comment:
This won't work, when using `CONCURRENT`, indexes that fail to create are left as invalid indexes, not deleted. Therefore you have to either remove and re-create the index, or reindex it.